### PR TITLE
TSGops -image (Light version of ADSWeb)

### PR DESCRIPTION
**Pull request by gupta1**
This PR fixes #
Added a new quality 'tsgops-image' for lighter version of adsweb - https://github.com/microsoft/azuredatastudio-release/pull/50
No python installation prompt for users when its already installed.
Hiding top menu action items and few extension items

### CODE REVIEW

File: extensions/notebook/src/jupyter/jupyterServerInstallation.ts
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/jupyterServerInstallation.ts:150]: Just have this be an `else if` and `else` on the original if/else. Don't need to nest it

File: src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts:25]: These should only be for the TSG ops quality for now, I don't think this is something we care about having for the non-TSG app. @kburtram ?

File: src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts:91]: Give this a name appropriate to what you're doing

File: src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts:95]: Would suggest a name that tells us what this list is for

File: extensions/notebook/src/jupyter/jupyterServerInstallation.ts
Comment (alanrenmsft) [inline extensions/notebook/src/jupyter/jupyterServerInstallation.ts:151]: is this good for all platforms?

File: src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts:92]: Please define the quality as a const in a common location.

Is there a reason you went for tsgops-image instead of just tsgops or something similar? Not a huge deal but seems unnecessarry to have it be that long

File: src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts:103]: This should only be registered when the quality is tsgops (and you can then remove the product.quality check in the contribution)

File: src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/extensions/browser/extensions.contribution.ts:23]: I would suggest putting this in its own file instead of overloading the extensions contribution. Make a ` src/sql/workbench/contrib/tsgops/browser/tsgops.contribution.ts` or something.

File: src/sql/workbench/contrib/extensions/browser/extensionsActions.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/extensions/browser/extensionsActions.ts:92]: This doesn't have anything to do with extensions so I'd prefer we name it something more generic like `workbench.action.hidePanel`

And for now since I don't see us using this anywhere else I'd prefer these also live in the tsgops.contribution.ts file I suggested making above.

File: src/sql/workbench/contrib/extensions/browser/extensionsActions.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/extensions/browser/extensionsActions.ts:122]: Instead of doing DOM manipulation why not just fine where the settings button is created and skip making it if the product quality is tsgops. That seems a lot simpler and less error prone.

(for example right now this assumes the aria-label is English which isn't necessarily true so it'll break on non-English installs)

File: src/sql/workbench/contrib/extensions/browser/extensionsActions.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/extensions/browser/extensionsActions.ts:139]: This is the same code as the contribution you made above (you just are never using the parameter you pass it in the contribution).

You should either make this only hide one container or only call it once in the contribution.

File: src/vs/workbench/browser/parts/activitybar/activitybarPart.ts
Comment (Charles-Gagnon) [inline src/vs/workbench/browser/parts/activitybar/activitybarPart.ts:757]: SQL CARBON EDIT

File: src/vs/workbench/browser/parts/titlebar/menubarControl.ts
Comment (Charles-Gagnon) [inline src/vs/workbench/browser/parts/titlebar/menubarControl.ts:45]: SQL CARBON EDIT

File: src/vs/workbench/services/activityBar/browser/activityBarService.ts
Comment (Charles-Gagnon) [inline src/vs/workbench/services/activityBar/browser/activityBarService.ts:33]: SQL CARBON EDIT

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: extensions/notebook/src/jupyter/jupyterServerInstallation.ts
Comment (Charles-Gagnon) [inline extensions/notebook/src/jupyter/jupyterServerInstallation.ts:150]: As @alanrenmsft pointed out - is this guaranteed to be accurate? Will this never be running on a windows machine?

File: src/sql/workbench/contrib/tsgops/tsgops.contribution.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/tsgops/tsgops.contribution.ts:18]: All of these actions should only be registered if the quality is tsgops

File: src/sql/workbench/contrib/tsgops/tsgops.contribution.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/tsgops/tsgops.contribution.ts:32]: You have the same keybinding for all of these, is that intended?

File: src/sql/workbench/contrib/tsgops/tsgops.contribution.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/tsgops/tsgops.contribution.ts:60]: Make a const value for this string

File: src/vs/workbench/workbench.common.main.ts
Comment (Charles-Gagnon) [inline src/vs/workbench/workbench.common.main.ts:522]: You should almost never be disabling eslint rules - and specifically you should never disable this one. Do you actually need to disable this? It should be fine...

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: extensions/notebook/src/jupyter/jupyterServerInstallation.ts
Comment (gupta1) [reply extensions/notebook/src/jupyter/jupyterServerInstallation.ts:150]: This is only planned to be used for adsweb which is based on linux

File: src/vs/workbench/workbench.common.main.ts
Comment (gupta1) [reply src/vs/workbench/workbench.common.main.ts:522]: It was giving me some warning. I will try to remove it and see

**Review by Charles-Gagnon — APPROVED** [review]
Please get approval from @kburtram before merging in

File: src/sql/workbench/contrib/tsgops/browser/tsgopsActions.ts
Comment (kburtram) [inline src/sql/workbench/contrib/tsgops/browser/tsgopsActions.ts:30]: nit: Windows UI guidelines says that "..." at end of command implies that the command will show a follow-up dialog/UX.  That doesn't seem to be the case here so I'm not sure why the command labels have a trailing "...".

File: src/sql/workbench/contrib/tsgops/browser/tsgopsActions.ts
Comment (kburtram) [inline src/sql/workbench/contrib/tsgops/browser/tsgopsActions.ts:48]: same comment of "..." here.  Also, the text doesn't really describe what the command is doing.  i.e. it hides more than the extension viewlet.

File: extensions/notebook/src/jupyter/jupyterServerInstallation.ts
Comment (kburtram) [inline extensions/notebook/src/jupyter/jupyterServerInstallation.ts:136]: this variable isn't really following normal TypeScript naming conventions, which would be `_tsgopsWeb` or similar.

**Review by kburtram — APPROVED** [review]
The normal way to customize the product for different SKU/bundles would be at build-time, rather than relying so much on runtime initialization.  But if this is the design you think works best for your needs then it looks fine.

**Pull request merged**
